pkg/reconciler/v1alpha1/revision/resolve.go

```diff
@@ -32,6 +34,40 @@ type digestResolver struct {
 	transport http.RoundTripper
 }
 
+const (
+	// Kubernetes CA certificate bundle is mounted into the pod here, see:
+	// https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/#trusting-tls-in-a-cluster
+	k8sCert = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+
+	// OpenShift cluster-created CA certificate bundle is mounted into the pod here, see:
+	// https://docs.openshift.com/online/dev_guide/secrets.html#service-serving-certificate-secrets
+	openShiftCert = "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"
+)
+
+// TODO: can we do this lazily so that we have a logger available?
+func newResolverTransport() *http.Transport {
+	roots, err := x509.SystemCertPool()
+	if err != nil {
+		roots = x509.NewCertPool()
+	}
+
+	paths := []string{k8sCert, openShiftCert}
+
+	for _, path := range paths {
+		if crt, err := ioutil.ReadFile(path); err != nil {
+			// Not fatal and we don't have a logger yet...
+		} else if ok := roots.AppendCertsFromPEM(crt); !ok {
+			// We don't have a logger yet :(
```

Comment:
Shall we make the `newResolverTransport` function return `(*http.Transport, err)`? An error should prohibit creation of the transport anyway I suppose?

The caller has a logger happily available.